themes/default/content/docs/intro/concepts/inputs-outputs.md

```diff
@@ -651,6 +651,339 @@ variables:
+                "Resource": worldFunction.arn, // Pulumi Resource Output
+                "End": true,
+            },
+        },
+    })
+});
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+state_machine = aws_native.stepfunctions.StateMachine("stateMachine",
+    role_arn=sfn_role.arn,
+    state_machine_type="EXPRESS",
+    definition_string=pulumi.Output.json_dumps({
+        "Comment": "A Hello World example of the Amazon States Language using two AWS Lambda Functions",
+        "StartAt": "Hello",
+        "States": {
+            "Hello": {
+                "Type": "Task",
+                "Resource": hello_function.arn, # Pulumi Resource Output
+                "Next": "World",
+            },
+            "World": {
+                "Type": "Task",
+                "Resource": world_function.arn, # Pulumi Resource Output
+                "End": True,
+            },
+        },
+    })
+)
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+{{% notes type="info" %}}
+The Pulumi Go SDK does not currently support Marshalling and Unmarshalling Maps with unknown values.
+It is tracked [here](https://github.com/pulumi/pulumi/issues/12460).
+
+The following is an example of JSON Serializing Pulumi Outputs in Go.
+{{% /notes %}}
+
+```go
+pulumi.JSONMarshal(pulumi.ToMapOutput(map[string]pulumi.Output{
+    "bool": pulumi.ToOutput(true),
+    "int":  pulumi.ToOutput(1),
+    "str":  pulumi.ToOutput("hello"),
+    "arr": pulumi.ToArrayOutput([]pulumi.Output{
+        pulumi.ToOutput(false),
+        pulumi.ToOutput(1.0),
+        pulumi.ToOutput(""),
+        pulumi.ToMapOutput(map[string]pulumi.Output{
+            "key": pulumi.ToOutput("value"),
+        }),
+    }),
+    "map": pulumi.ToMapOutput(map[string]pulumi.Output{
+        "key": pulumi.ToOutput("value"),
+    }),
+    // The following functionality is currently unsupported as myResource
+    // is an unknown value.
+    "unknown": myResource.ApplyT(func(res interface{}) (interface{}, error) {
+        return "Hello World!", nil
+    }),
+}))
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var stateMachine = Pulumi.Output.JsonSerialize(Output.Create(new {
+        Comment = "A Hello World example of the Amazon States Language using two AWS Lambda Functions",
+        StartAt = "Hello",
+        States = new Dictionary<string, object?>{
+        ["Hello"] = new {
+            Type = "Task",
+            Resource = helloFunction.Arn, // Pulumi Resource Output
+            Next = "World",
+        },
+        ["World"] = new {
+            Type = "Task",
+            Resource = worldFunction.Arn, // Pulumi Resource Output
+            End = true,
+        },
+    },
+}));
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
```

Comment:
This example does a bit more than what the title suggests it should.
We just want a JSON-parsed `Output<object>` from an `Output<string>`.
Although I'm not certain how you would demonstrate "this is an object" without doing stuff with it, so I guess this is fine.